sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -231,9 +256,26 @@ export class StudentControllerService {
       disabilityStatus: student.disabilityStatus,
       validSin: student.sinValidation.isValidSIN,
     };
-
+    if (options?.withAdditionalSpecificData && options?.withSensitiveData) {
```

Comment:
The conditions should add the extra properties without extra dependencies. For instance, if the `withAdditionalSpecificData` is true and `withSensitiveData` is false the entire block will be ignored.
Can you check if the below would work?
```ts
let specificData: Pick<
  AESTStudentProfileAPIOutDTO,
  "hasRestriction" | "identityProviderType"
>;
if (options?.withAdditionalSpecificData) {
  const studentRestrictions =
    await this.studentRestrictionService.getStudentRestrictionsById(
      studentId,
      {
        onlyActive: true,
      },
    );
  specificData = {
    hasRestriction: !!studentRestrictions.length,
    identityProviderType: student.user.identityProviderType,
  };
}
let sensitiveData: Pick<AESTStudentProfileAPIOutDTO, "sin">;
if (options?.withSensitiveData) {
  sensitiveData = {
    sin: student.sinValidation.sin,
  };
}
return {
  ...studentProfile,
  ...specificData,
  ...sensitiveData,
};
```

Reply:
Currently, we do not have a situation where `withAdditionalSpecificData` is true and `withSensitiveData` is false is possible. IF `withAdditionalSpecificData` is true, `withSensitiveData` will be true as per the current standings. But I get your point: Incrementally adding the properties for each of the passed parameters like the `withSensitiveData`.

Thanks. This approach makes more sense. 👍